core/src/main/java/org/javarosa/core/api/ClassNameHasher.java

```diff
@@ -0,0 +1,25 @@
+package org.javarosa.core.api;
+
+import org.javarosa.core.util.externalizable.Hasher;
+
+
+/**
+ * Most simple possible "hasher" - just gets the byte representation of the hash name
+ *
+ * @author wspride
+ */
+public class ClassNameHasher extends Hasher {
+
+    private final static int CLASS_HASH_SIZE = 32;
```

Comment:
Hmm... the main issue would be that a lot of the code as currently structured relies on know the tag size ahead of time. Probably we could rewrite a lot of this to be dynamic, but since this is the only hasher that would uses variable length not sure if that'd be worth it.